06.bowling_object/frame.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+require_relative 'shot'
+
+class Frame
+  LAST_FRAME = 9
+
+  attr_reader :shots
+
+  def initialize(frame_idx, marks)
+    @shots = []
+    @frame_idx = frame_idx
+    marks.each do |mark|
+      @shots << Shot.new(mark)
+      break if shots_completed?
+    end
+  end
+
+  def score(frames)
+    @shots.sum(&:score) + bonus_score(frames)
+  end
+
+  private
+
+  def shots_completed?
+    if @frame_idx < LAST_FRAME
+      strike? || @shots.size == 2
+    else
+      @shots.size == (strike? || spare? ? 3 : 2)
+    end
+  end
+
+  def bonus_score(frames)
+    next_frame = frames[@frame_idx + 1]
+    second_next_frame = frames[@frame_idx + 2]
+
+    return 0 if @frame_idx == LAST_FRAME
```

Comment:
```suggestion
    return 0 if @frame_idx == LAST_FRAME
    
    next_frame = frames[@frame_idx + 1]
    second_next_frame = frames[@frame_idx + 2]

```

最終フレームだと`next_frame`は`second_next_frame`は使われないので、最初に早期リターンしてしまう方がよいです

Reply:
```ruby
    next_frame, second_next_frame = frames[@frame_idx + 1, 2]
```

参考）あと、こんなふうに1行で書く方法もありますが、ちょっとトリッキーなので、元のコードのままで全然大丈夫です